### PR TITLE
added abacus interface, fixed a bug in reading cell in qe/pw/scf.

### PR DESCRIPTION
**Pull request by Liu-RX**
ABACUS plane waves scf interface is added in the folder 'dpdata/abacus'. 
Corresponding unittest is added in the folder 'tests/abacus.scf'.
A unit convert bug is fixed in reading cell function in qe/pw/scf.

### CODE REVIEW

**Comment by Liu-RX** [comment]
Not all of the files are necessary. I will delete the unnecessary ones.

发件人：Han Wang ***@***.***>
发送日期：2021-05-10 08:13:29
收件人：deepmodeling/dpdata ***@***.***>
抄送人：Liu-RX ***@***.***>,Author ***@***.***>
主题：Re: [deepmodeling/dpdata] added abacus interface, fixed a bug in reading cell in qe/pw/scf. (#153)

@amcadmus commented on this pull request.
Are all the files in the tests/abacus.scf/OUT.* directories needed for the unit test?

**Comment by Liu-RX** [comment]
I've read these errors. There are two sorts of errors. One of them is due to the none existence of an necessary output file of ABACUS. The other one happens in test_qe_pw_scf_energy_bug.py, which I did not run in my test. It is because the input script in the test takes 'crystal' unit but dpdata actually only takes Angstrom unit input. This problem is not noticed before, but I added a raise error if the input script is not in Angstrom unit, which triggers off in this test. I will fix this problem.

发件人：Han Wang ***@***.***>
发送日期：2021-05-10 08:34:25
收件人：deepmodeling/dpdata ***@***.***>
抄送人：Liu-RX ***@***.***>,Author ***@***.***>
主题：Re: [deepmodeling/dpdata] added abacus interface, fixed a bug in reading cell in qe/pw/scf. (#153)

Please notice the failures in the UTs

File: dpdata/qe/scf.py [hunk 1]
```diff
@@ -28,25 +28,48 @@ def get_cell (lines) :
...
+                coord.append([float(jj) for jj in ii.split()[1:4]])
+                atom_symbol_list.append(ii.split()[0])
+            coord = np.array(coord)
+        elif 'ATOMIC_POSITIONS' in iline and 'crystal' in iline.lower():
+            blk = get_block(lines, 'ATOMIC_POSITIONS')
+            for ii in blk:
+                coord.append([float(jj) for jj in ii.split()[1:4]])
+                atom_symbol_list.append(ii.split()[0])
+            coord = np.array(coord)
+            coord = np.matmul(coord, cell.T)
```
Comment (amcadmus) [inline dpdata/qe/scf.py:72]: I do not think we have the UTs for `"crystal"`, could you please add the UT for it? Thank you!

File: dpdata/system.py [hunk 2]
```diff
@@ -1231,6 +1233,11 @@ def from_qe_pw_scf(self, file_name) :
             = dpdata.qe.scf.get_frame(file_name)
         self.rot_lower_triangular()
 
+    @register_from_funcs.register_funcs('abacus/scf')
+    def from_qe_pw_scf(self, file_name) :
```
Comment (amcadmus) [inline dpdata/system.py:1237]: This definition is overwritting line 1224. Please give the function a different name, for example `from_abacus_scf`

**Review by amcadmus — CHANGES_REQUESTED** [review]

**Comment by Liu-RX** [comment]
Got it. I didn't notice the second problem before. Sorry for my poor coding... I'll fix them before next push and pr.

From: Han Wang ***@***.***>
Date: 2021-05-10 14:05:37
To: deepmodeling/dpdata ***@***.***>
Cc: Liu-RX ***@***.***>,Author ***@***.***>
Subject: Re: [deepmodeling/dpdata] added abacus interface, fixed a bug in reading cell in qe/pw/scf. (#153)

@amcadmus requested changes on this pull request.
In dpdata/qe/scf.py:
> +        elif 'ATOMIC_POSITIONS' in iline and 'crystal' in iline.lower():
+            blk = get_block(lines, 'ATOMIC_POSITIONS')
+            for ii in blk:
+                coord.append([float(jj) for jj in ii.split()[1:4]])
+                atom_symbol_list.append(ii.split()[0])
+            coord = np.array(coord)
+            coord = np.matmul(coord, cell.T)
I do not think we have the UTs for "crystal", could you please add the UT for it? Thank you!
In dpdata/system.py:
> @@ -1231,6 +1233,11 @@ def from_qe_pw_scf(self, file_name) :
             = dpdata.qe.scf.get_frame(file_name)
         self.rot_lower_triangular()
 
+    @register_from_funcs.register_funcs('abacus/scf')
+    def from_qe_pw_scf(self, file_name) :
This definition is overwritting line 1224. Please give the function a different name, for example from_abacus_scf

**Pull request closed without merging**